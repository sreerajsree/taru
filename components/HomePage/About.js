import AboutImage from './AboutImage';
import AboutText from './AboutText';

const transition = {
  ease: 'easeOut',
  duration: 0.5,
};

export default function About({ content, desktopImage, mobileImage }) {
  const aboutImageDesktop = `home/${desktopImage}`;
  const aboutImageMobile = `home/${mobileImage}`;
  return (
    <section id="about">
      <div className="flex w-full my-0 md:py-32 flex-col md:flex-row">
        <AboutImage image={aboutImageDesktop} transition={transition} />
        <AboutText image={aboutImageMobile} transition={transition} content={content} />
      </div>
    </section>
  );
}
