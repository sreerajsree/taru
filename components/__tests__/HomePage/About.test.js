import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

import About from '../../HomePage/About';

jest.mock('react-use');

describe('About', () => {
  beforeEach(() => {
    mockAllIsIntersecting(true);
  });
  test('renders and matches snapshot', () => {
    const { asFragment } = render(<About content="<p>some html</p>" desktopImage="ameira.jpg" mobileImage="ameira-garden.jpg" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('the AboutText is rendered', () => {
    render(<About content="<p>I have been a tour guide</p>" desktopImage="ameira.jpg" mobileImage="ameira-garden.jpg" />);
    expect(screen.getByText(/I have been a tour guide/)).toBeInTheDocument();
  });
  test('the AboutImage is rendered', () => {
    render(<About content="<p>I have been a tour guide</p>" desktopImage="ameira.jpg" mobileImage="ameira-garden.jpg" />);
    expect(screen.getByTestId('about-image')).toBeInTheDocument();
  });
});
