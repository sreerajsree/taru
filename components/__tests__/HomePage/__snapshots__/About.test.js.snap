// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders and matches snapshot 1`] = `
<DocumentFragment>
  <section
    id="about"
  >
    <div
      class="flex w-full my-0 md:py-32 flex-col md:flex-row"
    >
      <div
        class="w-full"
      >
        <div
          class="w-full h-full relative"
          style="opacity: 0; transform: translateX(-700px) translateZ(0);"
        >
          <img
            alt="Ameira"
            class="hidden md:block absolute z-0 object-cover w-overlap h-overlap max-w-overlap"
            data-testid="about-image"
          />
        </div>
      </div>
      <div
        class="w-full z-10 pb-10"
      >
        <div
          class="pt-12 mt-5 md:mt-12 w-full bg-grey-custom px-10 pb-10 h-full z-10"
          style="opacity: 0; transform: translateX(700px) translateZ(0);"
        >
          <h2
            class="text-6xl font-title mb-2 md:mb-0"
          >
            About Me
          </h2>
          <img
            alt="Ameira"
            class="md:hidden object-cover mb-2 md:mb-2"
          />
          <div
            class="about"
          >
            <p>
              some html
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
